### PR TITLE
XW-2486 | Force edit summary preference not checked when creating new article in VE

### PR DESCRIPTION
**Pull request by rybmat**
https://wikia-inc.atlassian.net/browse/XW-2486

//cc @Wikia/x-wing

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 21 issues
* MAJOR 16 major
* INFO 5 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.init.mw.ViewPageTarget.js](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js): File "ve.init.mw.ViewPageTarget.js" has 1822 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L18](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L18): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L68](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L68): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L111](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L111): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L201](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L201): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L415](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L415): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L433](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L433): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L544](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L544): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L766](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L766): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L835](https://github.com/Wikia/app/blob/6706f55b72cbd69918e55a1d60bcd2554cbc49ad/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L835): Remove this usage of alert(...). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1442)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 21 issues
* MAJOR 16 major
* INFO 5 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.init.mw.ViewPageTarget.js](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js): File "ve.init.mw.ViewPageTarget.js" has 1822 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L18](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L18): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L68](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L68): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L111](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L111): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L201](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L201): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L415](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L415): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L433](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L433): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L544](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L544): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L766](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L766): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L835](https://github.com/Wikia/app/blob/e49bc676192135bd5ddbeebee1f1cc25425f724f/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L835): Remove this usage of alert(...). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1442)

File: extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js [hunk 1]
```diff
@@ -986,6 +982,12 @@ ve.init.mw.ViewPageTarget.prototype.saveDocument = function ( saveDeferred ) {
 	}
 };
 
+ve.init.mw.ViewPageTarget.prototype.isSummaryMissing = function ( saveOptions ) {
+	return mw.user.options.get( 'forceeditsummary' ) &&
```
Comment (kvas-damian) [inline extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js]: This option comes as string value. In original code we're casting it to integer value.
I think we should keep this `+` to make it work correctly in all cases.

File: extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js [hunk 2]
```diff
@@ -986,6 +982,12 @@ ve.init.mw.ViewPageTarget.prototype.saveDocument = function ( saveDeferred ) {
 	}
 };
 
+ve.init.mw.ViewPageTarget.prototype.isSummaryMissing = function ( saveOptions ) {
```
Comment (kvas-damian) [inline extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js]: This method name is misleading. Can we think of better name?
For example: `shouldShowMissingSummaryMessage`

**Comment by WikiaTech** [comment]
SonarQube analysis reported 21 issues
* MAJOR 16 major
* INFO 5 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ve.init.mw.ViewPageTarget.js](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js): File "ve.init.mw.ViewPageTarget.js" has 1822 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L18](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L18): Make this function anonymous by removing its name: 'function() {...}'. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L68](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L68): Restrict what this loop acts on by testing each property. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AForIn)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L111](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L111): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L201](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L201): Take the required action to fix the issue indicated by this comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1134)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L415](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L415): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L433](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L433): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L544](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L544): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L766](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L766): Reduce the number of conditional operators (8) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. MAJOR [ve.init.mw.ViewPageTarget.js#L835](https://github.com/Wikia/app/blob/4251530229ef55eded480e5cba47ad75c48641c7/extensions/VisualEditor/modules/ve-mw/init/targets/ve.init.mw.ViewPageTarget.js#L835): Remove this usage of alert(...). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1442)

**Review by kvas-damian — APPROVED** [review]

**Review by latata — APPROVED** [review]

**Pull request merged**